### ISSUE TITLE
Accept all frame formats in the audio source

### ISSUE THREAD

**Issue by blinry** [nordlicht/nordlicht#26]
@nikoli suggested to use a video's audio, as well. It could be displayed as a [Moodbar](https://en.wikipedia.org/wiki/Moodbar) or a [Spectrogram](https://en.wikipedia.org/wiki/Spectrogram).

**Comment by rohieb** [comment]
@blinry: now that Debian has moved to ffmpeg again, you could give it another try.

**Comment by cinemast** [comment]
I tried it last week in Debian Testing and it worked just fine.

**Comment by rohieb** [comment]
Yeah, can also not reproduce this bug on current Debian testing, looks just like in your screenshot.

**Comment by rohieb** [comment]
Hm. Tried a bunch of videos now, none of them failed with spectrogram. @blinry, do you still know which audio format you were using when you encountered the bug?

**Comment by blinry** [comment]
I think this was in part caused by libav, where some decoders seemed to produce different sample formats (I think it happened on a version of *Sintel* with an AAC audio track, for example). The point is that it's definitely *possible* for FFmpeg to produce frame formats other than `AV_SAMPLE_FMT_FLTP`.

On my machine, it also happens with various flac-encoded files ([like this one](https://archive.org/download/MusopenCollectionAsFlac/Bach_GoldbergVariations/JohannSebastianBach-05-GoldbergVariationsBwv.988-Variation4.flac)), for which the decoder returns the format `AV_SAMPLE_FMT_S16` ([here is a list of all formats](https://www.ffmpeg.org/doxygen/2.5/group__lavu__sampfmts.html#ggaf9a51ca15301871723577c730b5865c5aea6132df57aebc3f76e10665395c46af)).

Huh, I'm also thinking that this does not really hinder a 1.0 release, as we will not make incompatible changes, just support more formats over time. But the documentation needs to be clear on this issue.
